lib/dry/transaction/step_adapters.rb

```diff
@@ -4,6 +4,16 @@ module Dry
   module Transaction
     class StepAdapters
       extend Dry::Container::Mixin
+
+      module Resolver
```

Comment:
I placed here this code iniside the `Resolver` module because it was been repeated across all `steps`. If we do not like this solution I have other options in my mind.

Instead of instantiating the step when registering, we could simply register the class,
later in the `call` in the step class, we could

```ruby
def call(input)
   args = [input] + Array(call_args)
   result = step_adapter.new(self, *args).call
   ......
end
```
This way we could make all the steps inherite from maybe some `StepBase` class that have this `resolve` method 😄